source/spec/controllers/urls_controller_spec.rb

```diff
@@ -0,0 +1,159 @@
+require 'rails_helper'
+
```

Comment:
Nice testing. Just as a head's up in cases where we have a gui in place, most cmm devs prefer to write feature tests. You'll see more testing of controllers and even more so the requests themselves in api builds.